### DISCUSSION TITLE
Code annotation in multi-figure doesn't not render

### DISCUSSION THREAD

**Discussion by snhansen** [quarto-dev/quarto-cli#7589] [Q&A] [answered]
### Description

I can't get code-annotation to work in code blocks that leads to multiple figures. If I remove the `layout-ncol: 1` options here, it does render it correctly, but I'd like to be able to specify either `layout-ncol: 1` or `layout-ncol: 2`. Not sure if it's a bug or if it's just me who are misunderstanding something. Note that removing the `fig-subcap` option also solves the issue.

	---
	format: html
	---

	```{r}
	#| layout-ncol: 1
	#| fig-cap: "Caption."
	#| fig-subcap:
	#|   - "Subcaption 2."
	#|   - "Subcaption 1."
	#| label: fig-label
	library(ggplot2)
	ggplot(data = mtcars, aes(y = mpg, x = wt)) + geom_point() # <1>
	ggplot(data = mtcars, aes(y = mpg, x = wt)) + geom_point()
	```
	1. Code-annotation.

**Comment by cscheid** [comment, marked as answer]
(This is a bug, thanks for catching it!)

@dragonstyle What's happening here is that the layout code has some clever bits that "yank out" the code cells from the layout output, and it's not pulling the definition list along with it.

I can add a hack to solve the problem, but I wonder if the right thing is for us to make code annotations part of the `DecoratedCodeBlock` AST node, so that the rest of our AST processing doesn't need to always remember that a CodeBlock can't be moved away by itself.

**Reply by mcanouil** [reply]
Closing in favour of:
- https://github.com/quarto-dev/quarto-cli/issues/7594
